src/routes/movies.ts

```diff
@@ -0,0 +1,352 @@
+ *             example:
+ *               - title: Movie 1
+ *                 description: Description 1
+ *                 releaseDate: 2022-01-01
+ *                 genre: Action
+ *               - title: Movie 2
+ *                 description: Description 2
+ *                 releaseDate: 2022-02-01
+ *                 genre: Drama
+ *       404:
+ *         description: Movies not found
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 error:
+ *                   type: string
+ *                   example: Movies not found
+ *       500:
+ *         description: Internal Server Error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 error:
+ *                   type: string
+ *                   example: Internal Server Error
+ */
+
+router.get('/', (req: Request, res: Response): void => {
+  movieModel
+    .find()
+    .then((movies: Movie[]): void => {
+      if (movies.length > 0) {
+        res.json(movies);
+      } else {
+        res.status(404).json({ error: 'Movies not found' });
+      }
+    })
+    .catch((error) => {
+      console.log(error);
+      res.status(500).json({ error: 'Internal Server Error' });
+    });
+});
+
+/**
+ * @openapi
+ * /movies:
+ *   post:
+ *     summary: Add a new movie
+ *     tags:
+ *       - Movies
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           example:
+ *             title: New Movie
+ *             description: New Movie Description
+ *             releaseDate: 2023-01-01
+ *             genre: Comedy
+ *     responses:
+ *       201:
+ *         description: Movie added successfully
+ *         content:
+ *           application/json:
+ *             example:
+ *               title: New Movie
+ *               description: New Movie Description
+ *               releaseDate: 2023-01-01
+ *               genre: Comedy
+ *       400:
+ *         description: Bad Request
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 error:
+ *                   type: string
+ *                   example: All fields are required
+ *       500:
+ *         description: Internal Server Error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 error:
+ *                   type: string
+ *                   example: Internal Server Error
+ */
+
+router.post('/', (req: Request, res: Response): Response | undefined => {
+  const { title, description, releaseDate, genre } = req.body;
+  if (!title || !description || !releaseDate || !genre) {
+    return res.status(400).json({ error: 'All fields are required' });
+  }
```

Comment:
When I pass 'releaseDate' in payload in a non-date format - it results in a 500 Internal error. Better to add data validation.